packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -257,8 +257,8 @@ export class SharedTree
 			moveToDetachedField(this.checkout.forest, cursor);
 			return {
 				schema: this.storedSchema.clone(),
-				tree: jsonableTreeFromFieldCursor(cursor),
-				removed: this.checkout.getRemovedRoots(),
+				tree: this.serializer.encode(jsonableTreeFromFieldCursor(cursor), this.handle),
+				removed: this.serializer.encode(this.checkout.getRemovedRoots(), this.handle),
```

Comment:
I don't think this is a good change. If you look at the type SharedTreeContentSnapshot, it (rather indorectly) uses handles, not encoded handles, so this makes this specific use of that type incorrectly typed.

Reply:
We talked about this, and went with this approach as the typing implies the output should be json compatible which only encoded handles are json compatible.

Reply:
If you want to change the type of the data here (which does seem reasonable), the typescript types used must also be changed to match.

Reply:
Note that "JsonableTree" is a specific tree format, not some arbitrary json compatible data. Infact, its not actually json compatible if you put handles in it since it extends NodeData which stores handles non-encoded

Reply:
I think we are un-opinionated, and we're totally clear on where and how this method was used. I think changing the types makes sense if we like this approach. If we don't like this approach we should move the decoding to another layer, likely in the stress test itself, which would probably require something like: https://github.com/microsoft/FluidFramework/compare/main...anthony-murphy:fuzz-validate-clients

Reply:
I have created https://github.com/microsoft/FluidFramework/pull/20812 to improve the JsonableTree docs.

Since packages/dds/shared-object-base/src/serializer.ts doesn't actually make any clear promises about the actual encoded format (just a lot of any), I don't think sending data which has gone through its encoding around is a good API. I think we are better off sticking to the in memory friendly format (with real handles) if practical.

If what motivated the change was tests issuescomparing handles, I think validateSnapshotConsistency might have a bug with not handling all cases of handles correctly (how to compare handles for equality isn't super clear, but that is using assert.deepEqual which might be wrong? Running the data through the encoder before it compares with deep equals might make sense?

Reply:
@jzaffiro i think the consensus here is move the handle decoding into the stress test/assert consistency function. i shared a branch above which i think makes that easier to do, as it provides more info to the assert consistency function, which should make creating a serializer per client easier.

Reply:
Based on the discussion, I think we can close [the ADO item](https://dev.azure.com/fluidframework/internal/_workitems/edit/7840) @jzaffiro created?

Reply:
Done!